include/ibamr/INSStaggeredHierarchyIntegrator.h

```diff
@@ -423,6 +422,11 @@ class INSStaggeredHierarchyIntegrator : public INSHierarchyIntegrator
 
     std::string d_U_P_bdry_interp_type = "LINEAR";
 
+    /*!
+     * Variables for graphical output.
+     */
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::NodeVariable<NDIM, double> > d_U_nc_var;
```

Comment:
1. Could we make the plotting variables all of type `Variable` and then decide data centerings at run time? This would mean there would be things like `U_plot_var` and `P_plot_var`, which could be the same as the true state variables if the data centerings match.
2. It would be good to have the option to output at least force and vorticity as nodal as well.
3. Pressure is naturally cell-centered, but for visualization it might be useful to have a nodal representation that is consistent with the coarse-fine interface treatment. E.g. if one wants to do isocontours/surfaces of pressure, those will be much cleaner as nodal until we get ghost cells into the VisIt output itself.

Reply:
There could / should be an operator to evaluate the node-centered curl of a staggered grid vector field; it probably is preferable to use that over evaluating the vorticity on another representation and then averaging to the nodes.

Reply:
Yes, definitely - lets handle the vorticity in a follow-up PR.